downstairs/src/main.rs

```diff
@@ -809,14 +793,122 @@ async fn proc(
                         }
                         fw.send(Message::LastFlushAck(last_flush)).await?;
                     }
-                    Some(msg) => {
-                        if !negotiated {
-                            bail!("expected HereIAm first");
+                    Some(Message::ExtentVersionsPlease) => {
+                        if negotiated != 3 {
```

Comment:
4?

Reply:
The upstairs does a 1-2-3-5 or 1-2-4-5, and I confused myself in thinking I needed it here as well.  Fixed in latest.